xray/commands/audit/scan.go

```diff
@@ -227,35 +233,44 @@ func (scanCmd *ScanCommand) performScanTasks(fileConsumer parallel.Runner, index
 	violations := []services.Violation{}
 	vulnerabilities := []services.Vulnerability{}
 	licenses := []services.License{}
+	flatResults := []services.ScanResponse{}
 	tempDirPath, err := fileutils.CreateTempDir()
 	if err != nil {
 		return false, err
 	}
 	for _, arr := range resultsArr {
 		for _, res := range arr {
-			if err = xrutils.WriteJsonResults(res, tempDirPath); err != nil {
-				return false, err
-			}
-			if scanCmd.printResults {
+			if scanCmd.printFormat == Table {
+				if err = xrutils.WriteJsonResults(res, tempDirPath); err != nil {
+					return false, err
+				}
 				violations = append(violations, res.Violations...)
 				vulnerabilities = append(vulnerabilities, res.Vulnerabilities...)
 				licenses = append(licenses, res.Licenses...)
+			} else if scanCmd.printFormat == Json {
+				flatResults = append(flatResults, *res)
 			}
 			if len(res.Violations) > 0 || len(res.Vulnerabilities) > 0 {
 				// A violation or vulnerability was found, the scan failed.
 				scanPassed = false
 			}
 		}
 	}
-	fmt.Println("The full scan results are available here: " + tempDirPath)
-	if len(violations) > 0 {
-		err = xrutils.PrintViolationsTable(violations, true)
-	}
-	if len(vulnerabilities) > 0 {
-		xrutils.PrintVulnerabilitiesTable(vulnerabilities, true)
-	}
-	if len(licenses) > 0 {
-		xrutils.PrintLicensesTable(licenses, true)
+	if scanCmd.printFormat == Table {
+		fmt.Println("The full scan results are available here: " + tempDirPath)
+		if len(violations) > 0 {
+			err = xrutils.PrintViolationsTable(violations, true)
+		}
+		if len(vulnerabilities) > 0 {
+			xrutils.PrintVulnerabilitiesTable(vulnerabilities, true)
+		}
+		if len(licenses) > 0 {
+			xrutils.PrintLicensesTable(licenses, true)
+		}
+	} else if scanCmd.printFormat == Json {
+		if len(flatResults) > 0 {
+			err = xrutils.PrintJson(flatResults)
+		}
```

Comment:
Let's print alway the results. It will make the automation easier.
```suggestion
		err = xrutils.PrintJson(flatResults)
```